### PR TITLE
Feature/displaying favorites ect

### LINKED ISSUE #12: Create Favorites Component/css file

### PR DESCRIPTION
**Pull request by ectweitmann**
## What Features are Being Added?
- Add functionality to the favorites view
- Add links to return a user to the home page
- Create component to handle bad paths called `Error404`
- Add a message to inform a user they haven't selected any favorites
- Add logo in place of h1
 - Add alt text to image covers
 
## What is Being Refactored?
- Refactored tabbing of `BookContainer`

## Where should the reviewer start?
- Multiple changes were made throughout all component files.
- Start in App to trace the changes to the Route paths

## How should this be manually tested?
- `git fetch` and npm start and navigate the app. All functionality should be up and running (favoriting, learn more button, error handling etc.)

## What are the relevant tickets?
- MVP app functionality for favoriting.

## Any Other Notes/Questions/Problems?
- Would like to go back and add responsive styling if possible.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -3,6 +3,9 @@
   <head>
     <meta charset="utf-8" />
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```
Comment (Fordo29) [inline public/index.html:7]: Nice find to match fonts

File: src/components/App.js [hunk 2]
```diff
@@ -23,24 +24,28 @@ class App extends Component {
...
      <section className='App'>
-       <Nav />
        <Switch>
         <Route exact path='/'>
+          <Nav />
           <BookContainer allBooks={this.state.books}/>
         </Route>
         <Route exact path='/favorites' render={() => {
-          return <Favorites />
+          return [<Nav location="favorites"/>, <Favorites />]
```
Comment (Fordo29) [inline src/components/App.js:40]: This was so neat to see that you can display an array of components

File: src/components/App.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ import React, { Component } from 'react';
 import BookContainer from './BookContainer';
 import Favorites from './Favorites';
 import BookDetails from './BookDetails';
+import Error404 from './Error404';
```
Comment (cagallo) [inline src/components/App.js:6]: I think it's smart to create this error page!

File: src/components/App.js [hunk 4]
```diff
@@ -23,24 +24,28 @@ class App extends Component {
...
+          <Nav />
           <BookContainer allBooks={this.state.books}/>
         </Route>
         <Route exact path='/favorites' render={() => {
-          return <Favorites />
+          return [<Nav location="favorites"/>, <Favorites />]
         }} />
-        <Route exact path='/:isbn' render={({ match }) => {
-          return <BookDetails isbn={match.params.isbn} />
+        <Route exact path='/:isbn/selectedBook' render={({ match }) => {
```
Comment (Fordo29) [inline src/components/App.js:42]: Nice way to learn a new way to limit the look up number.. corrects some error handling.

File: src/components/App.js [hunk 5]
```diff
@@ -23,24 +24,28 @@ class App extends Component {
...
+          return [<Nav />, <BookDetails isbn={match.params.isbn} />]
         }} />
+        <Route>
```
Comment (krdurrett) [inline src/components/App.js:45]: Great job with error handling

File: src/components/Error404.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import '../styles/Error404.css';
+
+const Error404 = () => {
```
Comment (krdurrett) [inline src/components/Error404.js:4]: Love the error handling component

File: src/components/Favorites.js [hunk 7]
```diff
@@ -1,20 +1,34 @@
...
+    apiCalls.getAllFavorites()
+      .then(booksData => {
+        const cleanedBooksData = booksData.map(book => cleanBookData(book));
+        this.setState({ favorites: cleanedBooksData})
+      })
+      .catch(error => this.setState({ error: error.message}))
+  }
+
+  render() {
+    return (
```
Comment (cagallo) [inline src/components/Favorites.js:25]: Clever using this shortcut for React fragments!

File: src/components/Favorites.js [hunk 8]
```diff
@@ -1,20 +1,34 @@
...
+      <>
```
Comment (krdurrett) [inline src/components/Favorites.js:26]: Nice conditional rendering

**Review by cagallo — APPROVED** [review]
That logo looks great! Fantastic work!

File: src/components/Nav.js [hunk 9]
```diff
@@ -1,12 +1,21 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
+import logo from '../Book_Club_Logo.png';
 import '../styles/Nav.css';
 
-const Nav = () => {
+const Nav = ({ location }) => {
+  let navLink;
+  if (location === 'favorites') {
+    navLink = <Link className="landing-page-link" to="/">Return Home »</Link>
```
Comment (krdurrett) [inline src/components/Nav.js:9]: This is a great idea for condition rendering of favorites or back to home button

**Review by krdurrett — APPROVED** [review]
Nice work on displaying favorites and on the conditional rendering/error handling

**Pull request merged**